tests/core/test_structured_data_signing.py

```diff
@@ -34,124 +34,263 @@
     if request.param == "text":
-        return {"text": structured_valid_data_json_string}
+        return {"text": eip712_example_json_string}
     elif request.param == "primitive":
-        return {"primitive": structured_valid_data_json_string.encode()}
+        return {"primitive": eip712_example_json_string.encode()}
     elif request.param == "dict":
-        return {"primitive": json.loads(structured_valid_data_json_string)}
+        return {"primitive": json.loads(eip712_example_json_string)}
     elif request.param == "hexstr":
-        return {"hexstr": structured_valid_data_json_string.encode().hex()}
+        return {"hexstr": eip712_example_json_string.encode().hex()}
+    else:
+        raise Exception("Unreachable")
+
+
+# --- EIP 712 example with added "cc" array field to Mail struct --- #
+
+@pytest.fixture
+def eip712_example_with_array_json_string():
+    return open("tests/fixtures/valid_eip712_example_with_array.json", "r").read()
+
+
+@pytest.fixture
+def eip712_example_with_array_types(eip712_example_with_array_json_string):
+    return json.loads(eip712_example_with_array_json_string)["types"]
+
+
+@pytest.fixture
+def eip712_example_with_array_domain_type(eip712_example_with_array_json_string):
+    return json.loads(eip712_example_with_array_json_string)["types"]["EIP712Domain"]
+
+
+@pytest.fixture
+def eip712_example_with_array_message(eip712_example_with_array_json_string):
+    return json.loads(eip712_example_with_array_json_string)["message"]
+
+
+@pytest.fixture
+def eip712_example_with_array_domain(eip712_example_with_array_json_string):
+    return json.loads(eip712_example_with_array_json_string)["domain"]
+
+
+@pytest.fixture(params=("text", "dict", "primitive", "hexstr"))
+def eip712_with_array_message_encodings(request, eip712_example_with_array_json_string):
+    if request.param == "text":
+        return {"text": eip712_example_with_array_json_string}
+    elif request.param == "primitive":
+        return {"primitive": eip712_example_with_array_json_string.encode()}
+    elif request.param == "dict":
+        return {"primitive": json.loads(eip712_example_with_array_json_string)}
+    elif request.param == "hexstr":
+        return {"hexstr": eip712_example_with_array_json_string.encode().hex()}
     else:
         raise Exception("Unreachable")
 
 
 @pytest.mark.parametrize(
-    'primary_type, expected',
+    'example, primary_type, expected',
     (
-        ('Mail', ('Person',)),
-        ('Person', ()),
+        ('eip712', 'Mail', ('Person',)),
+        ('eip712', 'Person', ()),
+        ('eip712_with_array', 'Mail', ('Person',)),
+        ('eip712_with_array', 'Person', ()),
     )
 )
-def test_get_dependencies(primary_type, types, expected):
-    assert get_dependencies(primary_type, types) == expected
+def test_get_dependencies(
+    example, primary_type, eip712_example_types, eip712_example_with_array_types, expected
+):
+    _types = (
```

Comment:
At one point, we were moving away from having `if` statements in our tests for this sort of branching logic. I think it's a little clearer to just test `eip712_with_array` in a separate test, but I don't have a strong opinion on that if you like this better.

Reply:
Agreed. I tried to make it so we can clearly see a path with `if` and basically an `elif` and otherwise purposely failing so that we don't get any false positives. I think this prevents the `if / else` from accidentally marking a test as `passed` which is arguably the main driver behind not having `if` statements in tests...

That said, I think separating it would be a lot better practice to be honest. I'll refactor this and it should resolve the comment on the `pytest.fail` as well. 👍